VerilogForVerificationStyle.md

```diff
@@ -0,0 +1,826 @@
+| --------------------------------- | -------------------------------------------- |
+| Virtual interfaces                | `virtual <if_name>_if <if_name>_vif;`        |
+| Environment                       | `<dut>_env env;`                             |
+| Env level config object           | `rand <dut>_env_cfg cfg;`                    |
+| Env coverage collection object    | `<dut>_env_cov cov;`                         |
+| Agent                             | `<dut>_agent m_<dut>_agent;`                 |
+| Agent level config object         | `rand <dut>_agent_cfg m_<dut>_agent_cfg;`    |
+| Agent coverage collection object  | `<dut>_agent_cov cov;`                       |
+| Driver                            | `<dut>_driver driver;`                       |
+| Monitor                           | `<dut>_monitor monitor;`                     |
+| Scoreboard                        | `<dut>_scoreboard scoreboard;`               |
+| Virtual sequencer                 | `<dut>_virtual_sequencer virtual_sequencer;` |
+| Sequencer                         | `<dut>_sequencer <dut>_sequencer;`           |
+
+The UVM testbench generation script will automatically create skeletons of these
+components with this naming scheme, so almost all of this naming is automated
+out.
+Additional naming guidelines:
+
+1. When instantiating any objects, the `name` argument to `type_id::create()`
+must match the name of the variable being assigned to.
+This ensures that any reporting messages refer to the right objects.
+2. To avoid name collisions, it is recommended that global types and types defined
+within a package have unique prefixes. This includes typedefs, enumerated
+types/values, package parameters, package localparams, any functions/tasks, and
+macros. Note that all package variables (parameters, etc.) must have a specified type.
+3. When instantiating a user-defined class member variable that is a class object:
+    *   Prefix the instance name of the class object with `m_`, as such:
+
+```systemverilog
+my_animal m_my_animal;
+```
+
+This scheme does not apply to any class member variables that are
+not themselves class objects.
+
+* If multiple instantiations of the class are desired without an array
+to store them, use `<class_name> m_<instance_name> to declare a variable,
+and use an informative instance name.
+
+```systemverilog
+my_animal m_my_cat;
+my_animal m_my_dog;
+my_animal m_my_fish;
+```
+
+* If an array of class handles is desired, choose a name such that it follows
+one of the following schemes:
+
+```systemverilog
+int NUM_ANIMALS = 10;
+// This plural convention is recommended, but both are equally acceptable
+my_animal my_animals[NUM_ANIMALS];
+my_animal my_animal_array[NUM_ANIMALS];
+```
+
+
+To avoid name collisions, it is recommended that global types and types defined
+within a package have unique prefixes. This include typedefs, enumerated
+types/values, package parameters, package localparams, any functions/tasks, and
+macros.
```

Comment:
This is repeated word for word from bullet 2 above - perhaps make the bullet above more concise such that this elaborates more on it.

Reply:
Oops, you're right, I missed deleting this duplicate. I added the detail about prefixing types defined with a class to the bullet 2 above.